lib/suncalc.js

```diff
@@ -0,0 +1,358 @@
+    result[time[2]] = fromJulian(Jset);
+  }
+
+  // Convert dates to the time at lat/lng
+  for (const time in result) {
+    const options = {
+      timeZone: tz,
+      year: "numeric",
+      month: "numeric",
+      day: "numeric",
+      hour: "numeric",
+      minute: "numeric",
+      second: "numeric",
+    };
+    result[time] = new Date(
+      Date.parse(
+        new Intl.DateTimeFormat("default", options).format(result[time])
+      )
+    );
+  }
+
+  return result;
+}
+
+// moon calculations, based on http://aa.quae.nl/en/reken/hemelpositie.html formulas
+
+function moonCoords(d) {
+  // geocentric ecliptic coordinates of the moon
+
+  const L = rad * (218.316 + 13.176396 * d); // ecliptic longitude
+  const M = rad * (134.963 + 13.064993 * d); // mean anomaly
+  const F = rad * (93.272 + 13.22935 * d); // mean distance
+  const l = L + rad * 6.289 * sin(M); // longitude
+  const b = rad * 5.128 * sin(F); // latitude
+  const dt = 385001 - 20905 * cos(M); // distance to the moon in km
+
+  return {
+    ra: rightAscension(l, b),
+    dec: declination(l, b),
+    dist: dt,
+  };
+}
+export function getMoonPosition(date, lat, lng) {
+  const lw = rad * -lng;
+  const phi = rad * lat;
+  const d = toDays(date);
+  const c = moonCoords(d);
+  const H = siderealTime(d, lw) - c.ra;
+  let h = altitude(H, phi, c.dec);
+  // formula 14.1 of "Astronomical Algorithms" 2nd edition by Jean Meeus (Willmann-Bell, Richmond) 1998.
+  const pa = atan(sin(H), tan(phi) * cos(c.dec) - sin(c.dec) * cos(H));
+
+  h = h + astroRefraction(h); // altitude correction for refraction
+
+  return {
+    azimuth: azimuth(H, phi, c.dec),
+    altitude: h,
+    distance: c.dist,
+    parallacticAngle: pa,
+  };
+}
+
+// calculations for illumination parameters of the moon,
+// based on http://idlastro.gsfc.nasa.gov/ftp/pro/astro/mphase.pro formulas and
+// Chapter 48 of "Astronomical Algorithms" 2nd edition by Jean Meeus (Willmann-Bell, Richmond) 1998.
+
+export function getMoonIllumination(date) {
+  const d = toDays(date || new Date()(Date.now()));
+  const s = sunCoords(d);
+  const m = moonCoords(d);
+  const sdist = 149598000; // distance from Earth to Sun in km
+  const phi = acos(
+    sin(s.dec) * sin(m.dec) + cos(s.dec) * cos(m.dec) * cos(s.ra - m.ra)
+  );
+  const inc = atan(sdist * sin(phi), m.dist - sdist * cos(phi));
+  const angle = atan(
+    cos(s.dec) * sin(s.ra - m.ra),
+    sin(s.dec) * cos(m.dec) - cos(s.dec) * sin(m.dec) * cos(s.ra - m.ra)
+  );
+
+  return {
+    fraction: (1 + cos(inc)) / 2,
+    phase: 0.5 + (0.5 * inc * (angle < 0 ? -1 : 1)) / Math.PI,
+    angle,
+  };
+}
+
+function hoursLater(date, h) {
+  return new Date(date.valueOf() + (h * dayMs) / 24);
+}
+
+// calculations for moon rise/set times are based on http://www.stargazing.net/kepler/moonrise.html article
+
+export function getMoonTimes(date, lat, lng, inUTC = false) {
+  const t = new Date(date);
+  const tz = tzlookup(lat, lng);
```

Comment:
here is the largest change I made, originally it was 

```javascript
if (inUTC) t.setUTCHours(0,0,0,0);
else t.setHours(0,0,0,0)
```
which would always set the time to the users system midnight and then try to find a moonrise and moonset in the next 24 hours. This would produce incorrect rises/sets for each day since the local timezone is different than user timezone. Now it checks for if the time is in UTC and if not calculates the local timezone offset from UTC and uses that to set midnight.